sxt/algorithm/base/accumulator.h

```diff
@@ -16,25 +16,16 @@
  */
 #pragma once
 
-#include "sxt/base/container/span.h"
-#include "sxt/execution/async/future_fwd.h"
+#include "sxt/base/macro/cuda_callable.h"
 
-namespace sxt::c21t {
-struct element_p3;
-}
-
-namespace sxt::basdv {
-class stream;
-}
-
-namespace sxt::mtxc21 {
+namespace sxt::algb {
 //--------------------------------------------------------------------------------------------------
-// async_compute_multiproduct
+// accumulator
 //--------------------------------------------------------------------------------------------------
-xena::future<> async_compute_multiproduct(basct::span<c21t::element_p3> products,
-                                          const basdv::stream& stream,
-                                          basct::cspan<c21t::element_p3> generators,
-                                          basct::cspan<unsigned> indexes,
-                                          basct::cspan<unsigned> product_sizes,
-                                          bool is_signed) noexcept;
-} // namespace sxt::mtxc21
+template <class T> class accumulator {
```

Comment:
Can you move this to multiexp/curve and require T to satisfy the curve concept? -- it's a bit too specific to put here

Reply:
Yes, that makes more sense. Updated.